### PR TITLE
Join sources with semicolon

### PR DESCRIPTION
**Pull request by pabloarosado**
* Improve the logic of how to join metadata sources (for example, use `;` without additional spaces when joining source names).
* Fix some bad source names in the electricity mix dataset.

### CODE REVIEW

File: etl/steps/data/garden/energy/2022-08-03/electricity_mix.meta.yml [hunk 1]
```diff
@@ -69,13 +69,13 @@ dataset:
...
     -
-      name: Our World in Data based on Ember's Global Electricity Review (2022).
+      name: Our World in Data based on Ember's Global Electricity Review (2022)
       published_by: Ember
       publication_year: 2022
       date_accessed: 2022-08-01
       url: https://ember-climate.org/data/data-catalogue/yearly-electricity-data/
     -
-      name: Our World in Data based on Ember's European Electricity Review (2022).
+      name: Our World in Data based on Ember's European Electricity Review (2022)
```
Comment (lucasrodes) [inline etl/steps/data/garden/energy/2022-08-03/electricity_mix.meta.yml:78]: Minor thing, but probably this does not belong in this PR?

**Review by lucasrodes — APPROVED** [review]

File: etl/steps/data/garden/energy/2022-08-03/electricity_mix.meta.yml (hunk 1 above)
Comment (Marigold) [reply etl/steps/data/garden/energy/2022-08-03/electricity_mix.meta.yml:78]: I think we're doing worse things than this :D.

**Review by Marigold — APPROVED** [review]

File: etl/steps/data/garden/energy/2022-08-03/electricity_mix.meta.yml (hunk 1 above)
Comment (pabloarosado) [reply etl/steps/data/garden/energy/2022-08-03/electricity_mix.meta.yml:78]: The thing is that I randomly picked this file to use as a test for the main changes of this PR :-).
